app/services/provider_authorisation_analysis.rb

```diff
@@ -0,0 +1,120 @@
+class ProviderAuthorisationAnalysis
+  attr_reader :permission, :auth, :application_choice, :course_option_id,
+              :training_provider, :ratifying_provider, :relationship
+
+  # This class is meant to be used after an auth call has already been performed
+  def initialize(permission:, auth:, application_choice:, course_option_id: nil)
```

Comment:
Do we ever want to pass in a course option id separately?

Reply:
Yes, when making or changing an offer an explicit `course_option_id` is required. So, if  `ProviderAuthorisationAnalysis` is used for `make_decisions` in this context, the course should be based on this `course_option_id`.

Reply:
See also: https://github.com/DFE-Digital/apply-for-teacher-training/pull/3273/commits/9c91f7fba93a1a3c2d03a80d2e02de2e4e80b4a1 (for `make_decisions`)